### PR TITLE
Merge changes from main

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: common/src/service/EventTraceActivity.cs [hunk 1]
```diff
@@ -35,21 +40,16 @@ public static EventTraceActivity Empty
                 return s_empty;
             }
         }
-
-        public static string Name => "E2EActivity";
-
-        // this field is passed as reference to native code.
-        public Guid ActivityId;
+        {
```
Comment (drwill-ms) [inline common/src/service/EventTraceActivity.cs:43]: will fix

File: iothub/device/src/HsmAuthentication/HttpClientHelper.cs [hunk 2]
```diff
@@ -19,19 +19,15 @@ internal static class HttpClientHelper
             Justification = "The caller of this method is in charge of disposing the HTTP client that is returned.")]
         public static HttpClient GetHttpClient(Uri providerUri)
         {
-            HttpClient client;
-
             if (providerUri.Scheme.Equals(HttpScheme, StringComparison.OrdinalIgnoreCase)
-                | providerUri.Scheme.Equals(HttpsScheme, StringComparison.OrdinalIgnoreCase))
+                || providerUri.Scheme.Equals(HttpsScheme, StringComparison.OrdinalIgnoreCase))
```
Comment (drwill-ms) [inline iothub/device/src/HsmAuthentication/HttpClientHelper.cs:23]: @timtay-microsoft here's an example of what I was talking about

Comment (timtay-microsoft) [reply iothub/device/src/HsmAuthentication/HttpClientHelper.cs:23]: Yeah that definitely seems accidental

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**